sdk/iot/Azure.Iot.Hub.Service/src/API Design.md

```diff
@@ -107,7 +107,65 @@ APIs for managing configurations for devices and modules.
 <details>
 
 ```csharp
+/// <summary>
+/// Gets a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configurationId">The unique identifier of the configuration.</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>TwinConfiguration for a automatic device/module management</returns>
+public virtual async Task<Response<TwinConfiguration>> GetConfigurationAsync(string configurationId, CancellationToken cancellationToken = default)
+
+/// <summary>
+/// Creates a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configuration">Twin configuration to create</param>
+/// <param name="cancellationToken">The cancellation token.</param>
+/// <returns>TwinConfiguration for a automatic device/module management</returns>
+public virtual async Task<Response<TwinConfiguration>> CreateConfigurationAsync(TwinConfiguration configuration, CancellationToken cancellationToken = default)
 
+/// <summary>
+/// Updates a configuration on the IoT Hub for automatic device/module management
+/// </summary>
+/// <param name="configurationId">The unique identifier of the configuration.</param>
```

Comment:
Isn't this already in the Configuration object below?

Reply:
This isn't resolved yet. The API should not take the configurationId and the TwinConfiguration object. It should just take the TwinConfiguration object since it has the configurationId

Reply:
same happened here, I did change it but got mismatch in the revert commit, will fix it soon